### PR TITLE
reproducers for gh-5060 and gh-5241

### PR DESCRIPTION
**Pull request by jsafarik**
//test: `@integrations-odata-read-update`

### CODE REVIEW

**Comment by syndesisqe** [comment]
Checkstyle verification failed with:
```
ui-tests/src/test/java/io/syndesis/qe/steps/other/
  ODataSteps.java:46:9: 'if' is not followed by whitespace. [WhitespaceAround]
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by avano** [comment]
retest this please

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
OData Connector | Read enum mapping from OData service tests 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

File: ui-tests/src/test/java/io/syndesis/qe/steps/other/ODataSteps.java [hunk 1]
```diff
@@ -42,8 +42,10 @@ public void insertEntityToODataService(final String entity, final String collect
     }
 
     @Then("^.*check that \"([^\"]*)\" entity in \"([^\"]*)\" collection contains$")
-    public void oDataServiceContains(final String entityKey, final String collection, DataTable table) {
-        // Currently supports only entityKey in number format
+    public void oDataServiceContains(String entityKey, final String collection, DataTable table) {
+        if (entityKey.matches("[A-Za-z]*")) {
```
Comment (asmigala) [inline ui-tests/src/test/java/io/syndesis/qe/steps/other/ODataSteps.java]: What is the purpose of this?

Comment (jsafarik) [reply ui-tests/src/test/java/io/syndesis/qe/steps/other/ODataSteps.java]: just very simple check if the entityKey contains characters, so it includes `'`entityKey`'`

Comment (asmigala) [reply ui-tests/src/test/java/io/syndesis/qe/steps/other/ODataSteps.java]: yeah, but why do we do that? If the entity key is an empty string or just letters of the alphabet, we quote it, and if it's something else, we don't?

Comment (jsafarik) [reply ui-tests/src/test/java/io/syndesis/qe/steps/other/ODataSteps.java]: if its just characters, we quote it so the odata service interprets it as string. If it is a number, we dont quote it. The check could be done differently, but this seems to work ok.

Comment (jsafarik) [reply ui-tests/src/test/java/io/syndesis/qe/steps/other/ODataSteps.java]: of course this is just a dumb check as strings containing both characters and numbers is not matching this. But for the purpose of our tests this works just fine.

Comment (asmigala) [reply ui-tests/src/test/java/io/syndesis/qe/steps/other/ODataSteps.java]: in fact it's me who's dumb, this is perfectly fine :D

**Pull request merged**